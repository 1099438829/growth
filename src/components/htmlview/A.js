import React, { Component, PropTypes } from 'react';
import { Text } from 'react-native';
import AppStyle from '../../theme/styles';
import hlper from '../../utils/helper';

class A extends Component {
  static componentName = 'A';

  static propTypes = {
    link: PropTypes.string,
    component: PropTypes.node,
  };

  static defaultProps = {
    link: null,
    component: null,
  };
  render() {
    return (<Text
      onPress={() => hlper.openLink(this.props.link)}
      style={AppStyle.a}
    >{this.props.component}</Text>);
  }
}
export default A;
