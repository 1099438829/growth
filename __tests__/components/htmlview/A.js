import { Text } from 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import A from '../../../src/components/htmlview/A';

it('renders correctly', () => {
  // eslint-disable-next-line no-unused-vars
  const a = renderer.create(
    <A link={'https://github.com/phodal/growth-ng'} component={<Text>2333</Text>} />,
  );
  const aJson = a.toJSON();
  expect(aJson).toMatchSnapshot();
});
