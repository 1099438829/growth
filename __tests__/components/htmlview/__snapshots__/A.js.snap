// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  onPress={[Function]}
  style={
    Object {
      "color": "#111",
      "textDecorationLine": "underline",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    2333
  </Text>
</Text>
`;
